### PR TITLE
SDIT-1318 Remove SYSTEM_USER from /api/offenders/{offenderNo}, /api/bookings/{bookingId} and /api/bookings/offenderNo/{offenderNo}

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java [hunk 1]
```diff
@@ -159,6 +159,7 @@ public Page<PrisonerBookingSummary> getPrisonerBookingsV2(
         @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
     @Operation(summary = "Offender detail.", description = "Offender detail.")
     @GetMapping("/{bookingId}")
+    @VerifyBookingAccess(overrideRoles = {"GLOBAL_SEARCH", "VIEW_PRISONER_DATA"})
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:162]: Pulled these into the Resource file (following on from Steve's work and also much easier to see permissions)

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java [hunk 2]
```diff
@@ -133,9 +133,10 @@ public class OffenderResource {
         @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
     @Operation(summary = "Full details about the current state of an offender")
     @GetMapping("/{offenderNo}")
+    @VerifyOffenderAccess(overrideRoles = {"GLOBAL_SEARCH", "VIEW_PRISONER_DATA"})
     public InmateDetail getOffender(
-        @RequestHeader(value = "version", defaultValue = "1.0", required = false) @Parameter(description = "Version of Offender details, default is 1.0, Beta is version 1.1_beta and is WIP (do not use in production)") final String version,
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:137]: Had to reorder to ensure offenderNumber was first param (for VerifyOffenderAccess)

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest.kt [hunk 3]
```diff
@@ -697,12 +767,29 @@ class BookingResourceIntTest : ResourceTest() {
...
         .expectBody()
+        .jsonPath("firstName").isEqualTo("ARTHUR")
+        .jsonPath("lastName").isEqualTo("ANDERSON")
         .jsonPath("profileInformation").isNotEmpty
         .jsonPath("physicalAttributes").isNotEmpty
         .jsonPath("physicalCharacteristics").isNotEmpty
         .jsonPath("physicalMarks").isNotEmpty
     }
 
+    @Test
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest.kt:778]: Added tests for separate flow

File: src/test/resources/features/booking_details.feature [hunk 4]
```diff
@@ -83,21 +83,11 @@ Feature: Booking Details
     When a basic offender booking request is made with booking id "-13"
     Then resource not found response is received from bookings API
 
-  Scenario: Request for specific offender as system user can return data even though booking is inactive
```
Comment (karenmillermoj) [inline src/test/resources/features/booking_details.feature:86]: Rewrote these two to check inactive bookings (the rest will need rewriting at a later time)

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**